interventions.py

```diff
@@ -1,45 +1,344 @@
+            for idx, batch in enumerate(batches):
+                batches[idx] = FLATTEN(batch)
+            self.batches = batches
+
+            # Permanently lock unused nodes
+            remaining_nodes = list(set([x for x in simulation.disease.graph.nodes]) - \
+                              set(FLATTEN(self.batches)))
+            for node in remaining_nodes:
+                simulation.disease.graph.nodes[node]['data']['occupant']['locked'] = True
+
+            self.current_batch = 0
+
+        # Animal moved out due to infection
+        for node in simulation.disease.graph.nodes:
+            if simulation.disease.graph.nodes[node]['data']['occupant']['state'] == \
+               simulation.disease.id_map['SY']:
+                move_out(simulation, node)
+
+        # Lock all empty rooms in non-active batches (even if they open up mid-run)
+        for idx, batch in enumerate(self.batches):
+            if idx != self.current_batch:
+                for node in batch:
+                    if simulation.disease.graph.nodes[node]['data']['occupant']['state'] == \
+                       simulation.disease.id_map['E']:
+                        simulation.disease.graph.nodes[node]['data']['occupant']['locked'] = True
+
+        # Perform batch move operations
+        if simulation.disease.time != 0 and simulation.disease.time % self.switch_time == 0:
+            self.current_batch = (self.current_batch + 1) % self.number_of_batches
+
+            # Move out oldest room and unlock
+            for node in self.batches[self.current_batch]:
+                move_out(simulation, node)
+                simulation.disease.graph.nodes[node]['data']['occupant']['locked'] = False
+
+        return self.swap_count
+
+class SnakeIntervention(object):
+    '''An intervention where a kennel ordering is determined and animals are
+    placed in that kennel order like a queue. Once an animal reaches the end of
+    the queue, they are removed from the simulation (placed in an independent area).
+    If an animal becomes infected, they are removed to an independent area.
+
+    Note: This intervention assumes that appropriate kennel order is the node order
+    (i.e. 1, 2, 3, 4 is the queue order).
+    '''
+
+    def __init__(self, name='', leave_time=72):
+        self.name = name
+        self.swap_count = 0
+        self.leave_time = leave_time
+
+    def update(self, simulation): #pylint: disable=W0613
+        '''Update the intervention given the simulation state.
+
+        Arguments:
+            simulation {Simulation} -- the current simulation state
+
+        Returns:
+            int -- the number of swaps updated
+        '''
+        get_node_state = lambda node: \
+            simulation.disease.graph.nodes[node]['data']['occupant']['state']
+        get_state_from_id = lambda id: \
+            simulation.disease.id_map[id]
+        node_ids = list(sorted([x for x in simulation.disease.graph.nodes], reverse=True))
```

Comment:
`sorted(...)` already returns a list, couldn't the outer `list(...)` be removed?